File: backend/app/models/income.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Numeric,
    Date,
    DateTime,
    ForeignKey,
)
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship

from app.db.database import Base


class Income(Base):
    __tablename__ = "income"

    id = Column(Integer, primary_key=True, index=True)

    user_id = Column(
        Integer,
        ForeignKey("users.id", ondelete="CASCADE"),
        nullable=False,
    )

    title = Column(String(150), nullable=False)

    amount = Column(Numeric(12, 2), nullable=False)

    source = Column(String(150), nullable=False)

    description = Column(String(500))

    payment_method = Column(String(50), nullable=False)

    income_date = Column(Date, nullable=False)

    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
    )

    user = relationship("User", back_populates="income")

File: frontend/src/types/income.ts
export interface Income {
  id: number;

  title: string;

  amount: number;

  source: string;

  description?: string;

  payment_method: string;

  income_date: string;

  user_id: number;

  created_at: string;
}

export interface CreateIncome {
  title: string;

  amount: number;

  source: string;

  description?: string;

  payment_method: string;

  income_date: string;
}